hummingbot/core/utils/eth_gas_station_lookup.py

```diff
@@ -28,6 +30,50 @@ def get_gas_price(in_gwei: bool = True) -> Decimal:
     return gas_price if in_gwei else gas_price / Decimal("1e9")
 
 
+def get_gas_limit(connector_name: str) -> int:
+    gas_limit = None
+    lookup = EthGasStationLookup.get_instance()
+    if connector_name not in lookup.gas_limits.keys():
+        gas_limit = request_gas_limit(connector_name)
+    else:
+        if connector_name == "balancer":
+            current_max_swaps = lookup.balancer_max_swaps
+            global_config_max_swaps = global_config_map["balancer_max_swaps"].value
+            if current_max_swaps == global_config_max_swaps:
+                gas_limit = lookup.gas_limits[connector_name]
+            else:
+                gas_limit = request_gas_limit(connector_name)
+                # update new max swap & gas limit due to config change
+                lookup.balancer_max_swaps = global_config_max_swaps
+                lookup.gas_limits = {connector_name: gas_limit}
+        elif connector_name == "uniswap":
+            gas_limit = lookup.gas_limits[connector_name]
```

Comment:
I think it's okay if we want to handle balancer specially for now. If that's the case, we may want to try something like:
```suggestion
        else:
            gas_limit = lookup.gas_limits[connector_name]
```
That way, it would be able to handle other ethereum connectors we would add to gateway later.
Also, we would certainly refactor this part of the code and `EthGasStationLookup` class later to be non-balancer specific. For instance, there's a possibility we would enhance Uniswap to go through a number of hops when it's routing swaps soon.Also, we can assume other connector have a max_swaps of 2.

Reply:
@sdgoh Let's talk through the changes to `eth_gas_station_lookup.py` after Asia standup. I think the current logic is close but could use a few tweaks.